gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/skis_equipped.mcfunction

```diff
@@ -0,0 +1,14 @@
+# Give skiing effects when ski requirements are met
+# @s = player meeting ski requirements
+# at @s
+# run from main
+
+# Give speed when wearing skis and on a snowy block
+effect give @s[predicate=gm4_mountaineering:on_snowy_block] speed 2 0 true
+
+# Give extra speed if descending
+execute at @a if block ~ ~-1.3 ~ minecraft:air run effect give @s speed 2 1 true
```

Comment:
The `@a` here creates issues, you probably meant `@s`; regardless, do this location setting in `main`.
The `if block air` check gives a lot of false positives when it comes to detecting falling, consider another method of checking for this -- think: what exactly do I want to detect? Players running down slopes?